src/components/UserImportModal/index.tsx

```diff
@@ -0,0 +1,213 @@
+
+   const updateData = (result) => {
+      setListUsers(result.data.splice(0, result.data.length - 1));
+   };
+
+   const handleValidateRow = (user: User) => {
+      let errKeys: string[] = [];
+      const isError = Object.keys(user).some((key) => {
+         switch (key) {
+            case 'username':
+               const checkUsername: boolean = !/^(?=[a-zA-Z0-9._]{5,20}$)/.test(
+                  user[key]
+               );
+               if (checkUsername) {
+                  errKeys.push(key);
+               }
+
+               return checkUsername;
+            case 'email':
+               const checkEmail: boolean =
+                  !/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
+                     user[key]
+                  );
+               if (checkEmail) {
+                  errKeys.push(key);
+               }
+               return checkEmail;
+            case 'password':
+               const checkPass: boolean = user[key].length < 6;
+               if (checkPass) {
+                  errKeys.push(key);
+               }
+               return checkPass;
+            case 'firstName' || 'lastName':
+               const checkName: boolean =
+                  !/^(?=[a-zA-Z0-9\u00C0-\u017F._]{2,20}$)/.test(
+                     nonAccentVietnameses(user[key])
+                  );
+               if (checkName) {
+                  errKeys.push(key);
+               }
+               return checkName;
+
+            case 'role':
+               const checkRole: boolean = ![
+                  textFromRole(Role.PM),
+                  textFromRole(Role.Member),
+                  textFromRole(Role.Admin),
+               ].includes(user[key]);
+               if (checkRole) {
+                  errKeys.push(key);
+               }
+               return checkRole;
+            case 'dateOfBirth':
+               if (user[key].length < 0) {
+                  errKeys.push(key);
+               }
+               return user[key].length < 0;
+            default:
+               return false;
+         }
+      });
+      return { isError, errKeys };
+   };
+
+   return (
+      <div className="userImport">
+         <div className="userImport__overlay" onClick={() => onClose()}></div>
+         <div className="userImport__container">
+            <h1>Import users</h1>
+            <div className="userImport__options">
+               <input
+                  type="file"
+                  accept=".csv, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"
+                  ref={inputRef}
+                  name="file"
+                  onChange={handleChange}
+               />
+               {dataFile && (
+                  <button onClick={handleDownLoadTemplate}>
+                     Download Template
+                  </button>
+               )}
```

Comment:
Nó là `template`, nghĩa là mẫu, để khi người ta không biết cấu trúc file csv như thế nào thì download về. E để khi chọn file mới hiện thì nó khá buồn cười.

Về heading, Ví dụ để import user e cần Username, email, first name, last name. Thì header nó phải bao gồm các trường đó để mình parse data ra. Như cái demo của papaparse mà e dùng nó có ví dụ đó, thì dòng đầu tiên là heading (Column 1,Column 2,Column 3,Column 4, E xem ví dụ của nó, thì tab string là đầu vào csv của mình, thì sau khi parse, data nó sẽ là array giống tab `Json to CSV` với mỗi object là 1 record để mình ghi vào db).